jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -515,6 +526,91 @@ static MethodHandle permuateHandleJEP370(MethodHandle methodHandle) {
 	}
 /*[ENDIF] Java14 */
 
+/*[IF Java15]*/
+    /**
+     * Generate a MethodHandle which translates:
+     *     FROM {VarHandle, Receiver, Intermediate ..., Value}ReturnType
+     *     TO   {Receiver, Intermediate ..., Value, VarHandle}ReturnType
+     *
+     * @param methodHandle to be permuted.
+     * @return the adapter MethodHandle which performs the translation.
+     */
+    static MethodHandle permuateHandleReferenceToJ9(MethodHandle methodHandle) {
```

Comment:
Creates a MethodHandle (MH) with MethodType (MT) for a VarHandle (VH) similar to the reference implementation and permutes/translates it to a MH/MT for an OpenJ9 VarHandle.